src/rviz/default_plugin/map_display.cpp

```diff
@@ -518,6 +580,59 @@ void MapDisplay::incomingUpdate(const map_msgs::OccupancyGridUpdate::ConstPtr& u
   Q_EMIT mapUpdated();
 }
 
+void MapDisplay::createSwatches()
+{
+  int width = current_map_.info.width;
+  int height = current_map_.info.height;
+  float resolution = current_map_.info.resolution;
+
+  int sw = width;
+  int sh = height;
+  int n_swatches = 1;
+
+  for(int i=0;i<4;i++){
+    ROS_INFO("Creating %d swatches", n_swatches);
+    for (unsigned i=0; i < swatches.size(); i++){
+        delete swatches[i];
```

Comment:
This is the only place that swatches are deleted, but the destructor of this class doesn't call this function directly nor does it call it indirectly that I can see.